src/pages/Edital/Detalhes/View.tsx

```diff
@@ -108,24 +114,38 @@ export default function EditalDetails() {
               alignItems="center"
               sx={{ width: "100%", my: 2 }}
             >
-              {edital?.arquivado ? (
-                <Typography sx={{ fontSize: 20, color: "primary.main" }}>
-                  Resultados disponíveis {/* link para resultado */}
-                </Typography>
+              {isTeacher ? (
```

Comment:
Acho que o melhor fuxo seria:
                
```
// edital arquivado
    -> resultados

// edital ñ arquivado
    // user
        // prof
          -> ver inscritos
        // ñ prof (aluno)
            // inscrito
              -> Você está inscrito
            // ñ inscrito
              -> Inscrever-se
    // não user
       -> Inscrever-se 
```
                       
Acho que dá pra fazer isso com um código +/- assim:

 ```
              {edital?.arquivado ? (
                <Typography sx={{ fontSize: 20, color: "primary.main" }}>
                  Resultados disponíveis {/* link para resultado */}
                </Typography>
              ) : (
                <Grid>
                  {user && isTeacher && (
                    <Button
                      type="button"
                      onClick={redirectToEnrolledList}
                      size="large"
                    >
                      Alunos Inscritos
                    </Button>
                  )}

                  {user && !isTeacher && isInscrito && (
                    <Typography sx={{ fontSize: 20, color: "primary.main" }}>
                      Inscrito(a)
                    </Typography>
                  )}

                  {!user ||
                    (user && !isTeacher && !isInscrito && (
                      <Button
                        type="button"
                        onClick={redirectToSubscribe}
                        size="large"
                      >
                        Inscreva-se
                      </Button>
                    ))}
                </Grid>
              )}
```

Reply:
Nesse caso o prof nao viria a lista de inscritos caso o PS estivesse arquivado? Fiz dessa for pra ele poder ver mesmo arquivado.

Reply:
Hummmm, entendi. Não tenho uma opinião formada sobre, mas de qualquer forma o "Resultados disponíveis também precisa aparecer pro prof, né?

E ainda assim você mexeu em outras partes da lógica que tava antes, né? Tem a questão do user e tudo mais

Acho que temos que ver melhor essa questão do prof ver a lista de inscritos caso o PS esteja arquivado, mas de qualquer forma acho que outras partes da lógica precisam ser refatoradas